test/release/examples/primers/interopWithC.chpl

```diff
@@ -0,0 +1,332 @@
+
+   writeln(sum(10, 10));
+   writeln(sum(10));
+// We must always make sure the types align as the C compiler specification allows for
+// different sizes for the same type depending on the compiler.
+
+// In order to pass an array to C, we use the following declaration:
+   extern proc sumArray(arr: [] int, size: c_int): c_int;
+//Where an array would be
+   var arr : [0..9] int = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
+// Keep in mind that since Chapel has 64 bit ints, the C function must also accept an
+// of type ``int64_t``
+   writeln("Sum of Array: ", sumArray(arr, 10:c_int));
+
+//
+// A C struct can be used in Chapel by declaring it as an ``extern record``.
+//
+
+   extern record rec {
+      var a: c_int;
+   }
+
+   extern proc giveRec(): rec; // returns an instance of the struct
+
+/*
+   You do not have to inform chapel about all the fields of a record,
+   only the ones that you wish to directly manipulate using Chapel code.
+
+   For example a record with no declared fields is possible even though
+   the actual C struct might have a nonzero number of fields.
+*/
+
+   extern record notReallyEmpty{
+   }
+
+// This means that the type is just reduced to being able to be passed around to
+// other functions and without the ability to directly manipulate it.
+
+// As of now the struct must be defined completely in the included header file
+// and must also be typdef'd.
+// In order to include a struct which is not typedef'd or if you want to import it
+// under another name simply state its external name after the ``extern`` keyword
+
+   extern "struct person" record person{
+      var name: c_string;
+      var age: c_int;
+   }
+
+// ``extern proc`` s can also be renamed in a similar fashion.
+//
+// Since most functions dealing with structs often return pointers, you can use the
+// ref intent for function arguments when their C counterparts are dealing with 
+// pointers
+
+   require "fact.c", "fact.h";
+
+   extern record data{
+      var x: c_int;
+   }
+
+   extern proc getNewData() ref : data;
+
+   ref d :data;
+   d = getNewData();
+
+   extern proc fact_d(ref x: data) ref : data;
+   ref f : data;
+   f = fact_d(d);
+
+   writeln(d);
+   writeln(f);
+
+// If you do not care about the type for a certain variable or argument, you can
+// use the ``opaque`` keyowrd to indicate to the compiler that you do not know about
+// the type.
+// Such a variable will not be much use except for the ability to pass it to different routines
+// which accept the same underlying type. (Be carefull here as it may lead to unmatched types)
+
+   extern proc getDataStructPtr(): opaque;
+   var structPtr: opaque = getDataStructPtr();
+
+// For the ability to use C code in Chapel without an external C file,
+// You can also use extern blocks which allow you to put C code directly into Chapel files.
+// To avoid cluttering you namespace you can also put these inside a module
+
+   module CDemo {
+      extern {
+         #include<math.h>
```

Comment:
I'dd put a space between #include and <math.h>.

Reply:
Putting those spaces in `fact.c` as well.